### PR TITLE
Add LARS support

### PR DESCRIPTION
**Pull request by typhoonzero**
Fix https://github.com/PaddlePaddle/Paddle/issues/6811

Related: https://github.com/PaddlePaddle/Paddle/pull/7788

To use, add `LARS_weight_decay=[some value greater than 0]` to enable LARS, LARS can also works along with current learning rate schedulers, like "polynomial_decay" e.g.

```python
opt = fld.optimizer.Momentum(
  learning_rate=layers.polynomial_decay(learning_rate=1.0, decay_steps=100, power=2.0),
  momentum=0.8,
  LARS_weight_decay=0.3)
```

or

```python
opt = fld.optimizer.Momentum(
  learning_rate=1.0,
  momentum=0.8,
  LARS_weight_decay=0.3)
```

### CODE REVIEW

File: python/paddle/fluid/layers/learning_rate_scheduler.py [hunk 1]
```diff
@@ -247,3 +248,41 @@ def piecewise_decay(boundaries, values):
                 tensor.assign(last_value_var, lr)
 
     return lr
+
+
+def append_LARS(params_grads, learning_rate, weight_decay):
+    """Applies LARS (LAYER-WISE ADAPTIVE RATE SCALING) to learning rate for
```
Comment (jacquesqiao) [inline python/paddle/fluid/layers/learning_rate_scheduler.py:256]: Maybe we can add the link to the paper here.

**Comment by jacquesqiao** [comment]
Please add a unit test for this learning rate scheduler strategy.

File: python/paddle/fluid/optimizer.py [hunk 2]
```diff
@@ -41,7 +41,10 @@ class Optimizer(object):
     but need to use one of it's implementation.
     """
 
-    def __init__(self, learning_rate, regularization=None):
+    def __init__(self,
+                 learning_rate,
+                 regularization=None,
+                 LARS_weight_decay=0.0):
```
Comment (jacquesqiao) [inline python/paddle/fluid/optimizer.py:47]: According to the paper, I think the default value of LARS_weight_decay may be 1.0.

**Comment by typhoonzero** [comment]
The program can not transpile to distributed version correctly when using LARS, still debugging.

**Comment by typhoonzero** [comment]
@jacquesqiao Can we merge this for now, so I can test it using NCCL2 distributed training

**Review by jacquesqiao — APPROVED** [review]
LGTM!
